Snakefile

```diff
@@ -469,8 +472,94 @@ rule fasta_headers:
             else print; }}' \
             {output.masked_consensus} > temp_{wildcards.sample}.fasta
         mv temp_{wildcards.sample}.fasta {output.masked_consensus}
+
+        """
+
+rule metadata_to_json:
+    input:
+        all_r1 = lambda wildcards: mapped[wildcards.sample][0],
+        all_r2 = lambda wildcards: mapped[wildcards.sample][1]
+    output:
+        temp("consensus_genomes/{reference}/{sample}.metadata.json")
+    shell:
+        """
+        python scripts/metadata_to_json.py {input.all_r1:q} {input.all_r2:q} > {output}
+        """
+
+rule masked_consensus_to_json:
+    input:
+        masked_consensus = rules.fasta_headers.output.masked_consensus
+    output:
+        temp("consensus_genomes/{reference}/{sample}.masked_consensus.json")
+    shell:
+        """
+        python scripts/fasta_to_json.py {input.masked_consensus} > {output}
         """
 
+rule summary_stats_to_json:
+    input:
+        bam_coverage = rules.bamstats.output.bamstats_file,
+        bowtie2 = rules.map.output.bt2_log
+    output:
+        temp("consensus_genomes/{reference}/{sample}.summary_stats.json")
+    shell:
+        """
+        python scripts/summary_stats_to_json.py --bamstats {input.bam_coverage} \
+            --bowtie2 {input.bowtie2} > {output}
+        """
+
+rule create_id3c_payload:
+    input:
+        metadata = rules.metadata_to_json.output,
+        masked_consensus = rules.masked_consensus_to_json.output,
+        summary_stats = rules.summary_stats_to_json.output
+    params:
+        status = 'complete'
+    output:
+        "consensus_genomes/{reference}/{sample}.payload.json"
+    shell:
+        """
+        python scripts/create_id3c_payload.py \
+            --masked-consensus {input.masked_consensus} \
+            --summary-stats {input.summary_stats} \
+            --metadata {input.metadata} \
+            --status {params.status} > {output}
+        """
+
+rule post_masked_consensus_and_summary_stats_to_id3c:
+    input:
+        rules.create_id3c_payload.output
+    params:
+        id3c_url = os.environ['ID3C_ASSEMBLY_ENDPOINT'],
+        id3c_username = os.environ['ASSEMBLY_ID3C_USERNAME'],
+        id3c_password = os.environ['ASSEMBLY_ID3C_PASSWORD']
+    log: "consensus_genomes/{reference}/{sample}.http-response.log"
+    run:
+        headers = {'Content-type': 'application/json'}
+        file = open(str(log), "w")
+
+        with open(str(input)) as f:
+            data = json.load(f)
+
+        try:
+            response = requests.post(
+                params.id3c_url,
```

Comment:
I think it will be better to do `urljoin(params.id3c_url, "/v1/receiving/consensus-genome")` instead of expecting the full endpoint in the environment variable. The code here in the Snakefile is written to the expectations of that particular `/v1` endpoint and so it's good to explicitly define that here rather than make it implicit.